### PR TITLE
ARROW-7812: [Packaging][Python] Use LLVM 8 in manylinux1 wheels

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

**Comment by kou** [comment]
@github-actions crossbow submit wheel-manylinux1-cp35m wheel-manylinux1-cp36m wheel-manylinux1-cp37m wheel-manylinux1-cp38

File: python/manylinux1/Dockerfile-x86_64_base [hunk 1]
```diff
@@ -20,7 +20,9 @@
 FROM quay.io/pypa/manylinux1_x86_64:2020-03-07-9c5ba95
 
 # Install dependencies
-RUN yum install -y xz ccache flex && yum clean all
+RUN yum install -y xz java-1.7.0-openjdk-devel ccache flex wget && yum clean all
```
Comment (pprudhvi) [inline python/manylinux1/Dockerfile-x86_64_base]: this file need not change

File: python/manylinux1/scripts/build_boost.sh [hunk 2]
```diff
@@ -32,7 +32,7 @@ popd
 pushd /arrow_boost
 ls -l
 ./bootstrap.sh
-./bjam -j${NCORES} dll-path="'\$ORIGIN/'" cxxflags='-std=c++11 -fPIC' cflags=-fPIC linkflags="-std=c++11" variant=release link=shared --prefix=/arrow_boost_dist --with-filesystem --with-date_time --with-system --with-regex install
+./bjam -j${NCORES} dll-path="'\$ORIGIN/'" cxxflags='-std=c++11 -fPIC' cflags=-fPIC linkflags="-std=c++11" variant=release link=static --prefix=/arrow_boost_dist --with-filesystem --with-date_time --with-system --with-regex install
```
Comment (pprudhvi) [inline python/manylinux1/scripts/build_boost.sh]: this change is not needed

**Comment by kou** [comment]
@github-actions crossbow submit wheel-manylinux1-cp35m wheel-manylinux1-cp36m wheel-manylinux1-cp37m wheel-manylinux1-cp38

File: python/manylinux1/Dockerfile-x86_64_base (hunk 1 above)
Comment (kou) [reply python/manylinux1/Dockerfile-x86_64_base]: Reverted.

File: python/manylinux1/scripts/build_boost.sh (hunk 2 above)
Comment (kou) [reply python/manylinux1/scripts/build_boost.sh]: Reverted.

**Comment by kou** [comment]
@github-actions crossbow submit wheel-manylinux1-cp35m wheel-manylinux1-cp36m wheel-manylinux1-cp37m wheel-manylinux1-cp38

**Comment by kou** [comment]
@github-actions crossbow submit wheel-manylinux1-cp35m wheel-manylinux1-cp36m wheel-manylinux1-cp37m wheel-manylinux1-cp38

**Comment by kou** [comment]
+1

**Pull request closed without merging**